CMakeLists.txt

```diff
@@ -0,0 +1,84 @@
+cmake_minimum_required(VERSION 2.8.12)
+
+cmake_policy(SET CMP0025 NEW)
+cmake_policy(SET CMP0028 NEW)
+cmake_policy(SET CMP0042 NEW)
+cmake_policy(SET CMP0048 NEW)
+cmake_policy(SET CMP0057 NEW)
+cmake_policy(SET CMP0076 NEW)
+
+list(APPEND CMAKE_MODULE_PATH
+     "${CMAKE_CURRENT_LIST_DIR}/cmake"
+     "${CMAKE_CURRENT_LIST_DIR}/cmake/packages"
+     "${CMAKE_CURRENT_LIST_DIR}/cmake/modules/share/modules/cmake")
+
+include(CMConfig)
+include(CMSetupVersion)
+
+if(NOT CMAKE_WORKSPACE_NAME OR NOT ("${CMAKE_WORKSPACE_NAME}" STREQUAL "actor"))
+    cm_workspace(actor)
+endif()
+
+macro(cm_find_package NAME)
+    foreach(ITERATOR ${CMAKE_WORKSPACE_LIST})
+        if(NOT "${NAME}" MATCHES "^${ITERATOR}_.*$" AND NOT "${NAME}" STREQUAL CM)
+            find_package(${ARGV})
+        else()
+            set(${ARGV0}_FOUND ON CACHE BOOL "")
+        endif()
+    endforeach()
+endmacro()
+
+cm_project(core WORKSPACE_NAME ${CMAKE_WORKSPACE_NAME} LANGUAGES C CXX)
+
+if(NOT Boost_FOUND AND NOT CMAKE_CROSSCOMPILING)
+    cm_find_package(Boost)
+endif()
+
+cm_find_package(CM)
+include(CMDeploy)
+include(FindPkgConfig)
+
+option(BUILD_WITH_CCACHE "Build with ccache usage" TRUE)
+option(BUILD_TESTS "Build unit tests" FALSE)
```

Comment:
Recently I've found interesting thing about tests in our repos. 
Let's make it as it should be. Replace `add_subdirectory` with `cm_add_test_subdirectory` and don't use `BUILD_TESTS`

Reply:
Changed.

Reply:
I thought I mentioned [this issue](https://github.com/NilFoundation/crypto3/issues/146), pardon.`cm_add_test_subdirectory` is still wrapped into BUILD_TEST condition, but it already has condition inside `cm_add_test_subdirectory`. Tests are enabled with `ENABLE_TESTS`

Reply:
Yes, sorry.